addons/stock/models/product.py

```diff
@@ -670,11 +670,12 @@ def action_open_product_lot(self):
         self.ensure_one()
         action = self.env.ref('stock.action_production_lot_form').read()[0]
         action['domain'] = [('product_id.product_tmpl_id', '=', self.id)]
+        action['context'] = {'products': self.product_variant_ids.ids, 'operator': 'in'}
```

Comment:
@sle we have passed dynamic domain because we want to show only same product when we open from the product.product but when we open lot view from product template we have to show all the products of template.

Now action domain will apply on the lot record not on the product domain.
we can't set static domain because operator will be change.

we have tried to put full domain in the context but it is not working.

if you have better solution then please provide us.

Reply:
operator will be change? why?

Reply:
@sle-odoo 

When we open from the template we need to show all the variants 
so domain will be ('id', 'in', variants) but when form open product ( variant ) it should be 
('id', '=', variant) ... to not show other variant 

is this possible by just domain change ?